src/mon/OSDMonitor.cc

```diff
@@ -434,7 +434,9 @@ void OSDMonitor::on_active()
     list<MOSDFailure*> ls;
     take_all_failures(ls);
     while (!ls.empty()) {
-      dispatch(ls.front());
+      MonOpRequestRef op =
+        mon->op_tracker.create_request<MonOpRequest>(ls.front());
+      dispatch(op);
       ls.pop_front();
```

Comment:
Hmm.  Is this better than keeping the OpRequest alive the whole time the mon is holding onto the MOSDFailure?  I forget how long that might be.